### PR TITLE
MOSIP-20644_Moving landing zone to Minio

### PR DESCRIPTION
**Pull request by JyotiPrakashNayak**

### CODE REVIEW

File: registration-processor/init/registration-processor-packet-receiver-stage/src/main/java/io/mosip/registration/processor/packet/receiver/service/impl/PacketReceiverServiceImpl.java [hunk 1]
```diff
@@ -506,8 +518,26 @@ public MessageDTO processPacket(File file) {
...
-				fileManager.put(packetId, new ByteArrayInputStream(encryptedByteArray),
-						DirectoryPathDto.LANDING_ZONE);
+				if(landingZoneType.equalsIgnoreCase(LandingZoneTypeConstant.DMZ_SERVER)) {
+					fileManager.put(packetId, new ByteArrayInputStream(encryptedByteArray),
+							DirectoryPathDto.LANDING_ZONE);
+				}
+				else if(landingZoneType.equalsIgnoreCase(LandingZoneTypeConstant.OBJECT_STORE)) {
+					 boolean result =objectStoreAdapter.putObject(landingZoneAccount, registrationId, null, null, packetId, encryptedInputStream);
+					 if(!result) {
+						 messageDTO.setInternalError(Boolean.TRUE);
```
Comment (MonobikashDas) [inline registration-processor/init/registration-processor-packet-receiver-stage/src/main/java/io/mosip/registration/processor/packet/receiver/service/impl/PacketReceiverServiceImpl.java]: instead of handling all values here its better to throw exception here and handle all exceptions in the process() method. In that case, we dont have to set these values individually in all methods. This is how we have been following till now.

File: registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/service/impl/PacketUploaderServiceImpl.java [hunk 2]
```diff
@@ -562,21 +569,26 @@ public int getMaxRetryCount() {
...
         List<String> pathSegment = new ArrayList<>();
         pathSegment.add(packetId + extention);
         byte[] packet = null;
 
         try {
+        	if(landingZoneType.equalsIgnoreCase(LandingZoneTypeConstant.DMZ_SERVER)) {
             packet = (byte[]) restClient.getApi(ApiName.NGINXDMZURL, pathSegment, "", null, byte[].class);
+        	}
+        	else if(landingZoneType.equalsIgnoreCase(LandingZoneTypeConstant.OBJECT_STORE)) {
+        	packet=IOUtils.toByteArray(objectStoreAdapter.getObject(landingZoneAccount, registrationId, null, null, packetId));
```
Comment (MonobikashDas) [inline registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/service/impl/PacketUploaderServiceImpl.java:583]: make sure objectStoreAdapter.getObject() exceptions are handled properly

File: registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/stage/PacketUploaderStage.java [hunk 3]
```diff
@@ -53,6 +54,8 @@ public class PacketUploaderStage extends MosipVerticleAPIManager {
 	@Value("${mosip.regproc.packet.uploader.message.expiry-time-limit}")
 	private Long messageExpiryTimeLimit;
 
+	@Value("${move.packets.to.ObjectStore.Period.milliseconds}")
+	private Integer movePacketsToObjectStorePeriod;
```
Comment (MonobikashDas) [inline registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/stage/PacketUploaderStage.java]: why did you add this config, i dont see any usage.

File: registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	private String landingZoneAccount;
+		
+	@Value("${landing.zone.type:ObjectStore}")
+	private String landingZoneType;
+	
+	@Value("${registration.processor.packet.ext}")
+    private String extention;
+	
+	@Value("${landing.zone.move.schedule:0 0 0 * * *}")
+    private static  String schedule;
```
Comment (MonobikashDas) [inline registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java]: was it agreed to use schedule as hardcoded value?? or it should be configuration>?

File: registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+	  *   
+	  * moves the packets from dmz server to objectstore if the landing zone has been changed to 
+      */
+	 @Scheduled(fixedDelayString = "${mosip.regproc.landing.zone.fixed.delay.millisecs:43200000}",
+	            initialDelayString = "${mosip.regproc.landing.zone.inital.delay.millisecs:300000}")
+	   public void movePacketsToObjectStore() {
+		if(landingZoneType.equalsIgnoreCase(LandingZoneTypeConstant.DMZ_SERVER)) {
+			regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(),
+					"", "PacketUploaderServiceImpl::movePacketsToObjectStore()::entry");
+			List<String> regIdList=registrationStatusService.getAllRegistrationIds();
```
Comment (MonobikashDas) [inline registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java]: There could be millions of packets so list will not even work here. Also I see an issue here. The scheduler will keep uploading packets to objectstore even if its uploaded before. It should be limited by number. In one run only limited number of packets should be processed not all. 
and dont read regidlist from the registration table. Insted use - 
apache fileutils.listFiles to list all files in a directory. Read first few and process them and delete them, Then read next batch like that.
Reference -
http://www.avajava.com/tutorials/lessons/how-do-i-get-all-files-with-certain-extensions-in-a-directory-including-subdirectories.html

File: registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+			for(String regId:regIdList) {
+				List<String> packetIdList=syncRegistrationService.getAllPacketIds(regId);
+				for(String packetId:packetIdList) {
+					 List<String> pathSegment = new ArrayList<>();
+				     pathSegment.add(packetId + extention);
+					
+						try {
+							byte[] packet = (byte[]) restClient.getApi(ApiName.NGINXDMZURL, pathSegment, "", null, byte[].class);
+							if(packet!=null) {
```
Comment (MonobikashDas) [inline registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java]: dont call restClient.getApi instead use fileutils.listfile().<get first 10> like that. 10 can be a configurable number.

File: registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/dao/RegistrationStatusDao.java [hunk 7]
```diff
@@ -118,7 +118,11 @@ public List<RegistrationStatusEntity> findAll(String rid) {
 
 		return registrationStatusRepositary.findByRegId(rid);
 	}
-
+	
+	public List<String> findAllIds() {
+		
+		return registrationStatusRepositary.findAllRegIds();
```
Comment (MonobikashDas) [inline registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/dao/RegistrationStatusDao.java]: remove this. we should not even have a method to get all rids. It may break our system

File: registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/repositary/RegistrationRepositary.java [hunk 8]
```diff
@@ -63,5 +63,8 @@ public List<String> getWithoutStatusCodes(@Param("regIds") List<String> regIds,
 
 	@Query(value ="SELECT * FROM registration r WHERE r.status_code =:statusCode  order by r.upd_dtimes LIMIT :fetchSize ", nativeQuery = true)
 	public List<RegistrationStatusEntity> getResumablePackets(@Param("statusCode") String statusCode,@Param("fetchSize") Integer fetchSize);
+
+	@Query("SELECT registration.regId FROM RegistrationStatusEntity registration")
```
Comment (MonobikashDas) [inline registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/repositary/RegistrationRepositary.java]: delete

File: registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/dao/SyncRegistrationDao.java [hunk 9]
```diff
@@ -210,4 +210,9 @@ public String getSaltValue(Long id) {
 		return saltEntity.getSalt();
 	}
 
+	public List<String> getAllPacketIds(String registrationId) {
+		// TODO Auto-generated method stub
```
Comment (MonobikashDas) [inline registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/dao/SyncRegistrationDao.java]: dellete

File: registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/RegistrationStatusServiceImpl.java [hunk 10]
```diff
@@ -129,6 +129,31 @@ public List<InternalRegistrationStatusDto> getAllRegistrationStatuses(String reg
 					PlatformErrorMessages.RPR_RGS_REGISTRATION_TABLE_NOT_ACCESSIBLE.getMessage(), e);
 		}
 	}
+	
+	@Override
+	public List<String> getAllRegistrationIds() {
+
+		regProcLogger.debug(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(),
```
Comment (MonobikashDas) [inline registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/RegistrationStatusServiceImpl.java]: delete

**Review by MonobikashDas — CHANGES_REQUESTED** [review]

File: registration-processor/registration-processor-landing-zone-Utility/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,85 @@
+FROM openjdk:11
```
Comment (MonobikashDas) [inline registration-processor/registration-processor-landing-zone-Utility/Dockerfile:1]: Since you are reading packets from LANDING_ZONE location, you have to do a volume mapping of LANDING_ZONE in dockerfile. Check the stage-group-1 dockerfile for same.

File: registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+			try {
+				InputStream stream=new FileInputStream(packet);
+				result = objectStoreAdapter.putObject(landingZoneAccount,regId, null, null, packetId, stream);
+				stream.close();
+			   if(!result) {
+					regProcLogger.error(LoggerFileConstant.SESSIONID.toString(), LoggerFileConstant.USERID.toString(),
+									 regId, packetId+":Packet store not accesible");
+				}
+				else {
+					fileManager.deletePacket(DirectoryPathDto.LANDING_ZONE, packetId);
```
Comment (MonobikashDas) [inline registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java:113]: Did you verify if delete is working? i remember we had simmilar functionality before which didnot work but not sure if we were using same fileManager.

**Review by MonobikashDas — CHANGES_REQUESTED** [review]

File: registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java (hunk 12 above)
Comment (JyotiPrakashNayak) [reply registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java:113]: It is working in local.

File: registration-processor/init/registration-processor-packet-receiver-stage/src/main/java/io/mosip/registration/processor/packet/receiver/service/impl/PacketReceiverServiceImpl.java (hunk 1 above)
Comment (JyotiPrakashNayak) [reply registration-processor/init/registration-processor-packet-receiver-stage/src/main/java/io/mosip/registration/processor/packet/receiver/service/impl/PacketReceiverServiceImpl.java]: I have modified the code.

File: registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/service/impl/PacketUploaderServiceImpl.java (hunk 2 above)
Comment (JyotiPrakashNayak) [reply registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/service/impl/PacketUploaderServiceImpl.java:583]: I have added few lines to handle it separately.

File: registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/stage/PacketUploaderStage.java (hunk 3 above)
Comment (JyotiPrakashNayak) [reply registration-processor/pre-processor/registration-processor-packet-uploader-stage/src/main/java/io/mosip/registration/processor/packet/uploader/stage/PacketUploaderStage.java]: I have removed it.

File: registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java (hunk 4 above)
Comment (JyotiPrakashNayak) [reply registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java]: I have removed the code.

File: registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java (hunk 5 above)
Comment (JyotiPrakashNayak) [reply registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java]: I have modified the code.

File: registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java (hunk 6 above)
Comment (JyotiPrakashNayak) [reply registration-processor/registration-processor-landing-zone-Utility/src/main/java/io/mosip/registration/processor/util/LandingZoneUtility.java]: I have modified the code.

File: registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/dao/RegistrationStatusDao.java (hunk 7 above)
Comment (JyotiPrakashNayak) [reply registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/dao/RegistrationStatusDao.java]: I have deleted the code.

File: registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/repositary/RegistrationRepositary.java (hunk 8 above)
Comment (JyotiPrakashNayak) [reply registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/repositary/RegistrationRepositary.java]: I have deleted the code.

File: registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/dao/SyncRegistrationDao.java (hunk 9 above)
Comment (JyotiPrakashNayak) [reply registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/dao/SyncRegistrationDao.java]: I have deleted the code.

File: registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/RegistrationStatusServiceImpl.java (hunk 10 above)
Comment (JyotiPrakashNayak) [reply registration-processor/registration-processor-registration-status-service-impl/src/main/java/io/mosip/registration/processor/status/service/impl/RegistrationStatusServiceImpl.java]: I have deleted the code.

File: registration-processor/registration-processor-landing-zone-Utility/Dockerfile (hunk 11 above)
Comment (JyotiPrakashNayak) [reply registration-processor/registration-processor-landing-zone-Utility/Dockerfile:1]: I have added the volume

**Review by MonobikashDas — APPROVED** [review]
